snuba/writer.py

```diff
@@ -34,3 +40,27 @@ def write(self, rows):
             'colnames': ", ".join(col.escaped for col in columns),
             'table': self.__schema.get_table_name(),
         }, [self.__row_to_column_list(columns, row) for row in rows], types_check=False)
+
+
+class HTTPBatchWriter(BatchWriter):
+    def __init__(self, schema, host, port, options=None):
+        self.__schema = schema
+        self.__base_url = 'http://{host}:{port}/'.format(host=host, port=port)
+        self.__options = options if options is not None else {}
+
+    def __default(self, value):
+        if isinstance(value, datetime):
+            return value.strftime(DATETIME_FORMAT)
+        else:
+            raise TypeError
+
+    def __encode(self, row):
+        return json.dumps(row, default=self.__default).encode('utf-8')
```

Comment:
do you actually need to encode it utf-8 after running this ? Or is it for Python 2.7?

Reply:
The transfer encoding requires the per-chunk length header to know the length (in bytes) of the payload: https://github.com/not-kennethreitz/requests/blob/9a2e5df000691ad28613524ca7f80aa28a94a8d5/requests/adapters.py#L470

I think there was some reason I needed to make this change, but I don't recall specifically what it was. I vaguely remember seeing this in the initial tests with the HTTP writer, which led to a2260f82962c48842779155b3ead934cf17ec4b2, but I can't find any surrounding context about the change on the prototype branch.

It doesn't actually make a difference for Python 2.7, but it does for Python 3.7, at least as far as the literal types go: 

```
Python 2.7.14 (default, Sep 25 2017, 09:53:22) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.37)] on darwin
>>> import json 
>>> json.dumps({'hi': u'\N{SNOWMAN}'})
'{"hi": "\\u2603"}'  # byte string, not unicode string (unicode should be escaped)
```

```
Python 3.7.2 (default, Feb 12 2019, 08:16:11) 
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
>>> import json
>>> json.dumps({'hi': u'\N{SNOWMAN}'})
'{"hi": "\\u2603"}'  # unicode string, not byte string (unicode should be escaped)
```

Reply:
mh, ok